tiatoolbox/models/engine/semantic_segmentor_new.py

```diff
@@ -0,0 +1,481 @@
+
+        Returns:
+            dict or Path or :class:`AnnotationStore`:
+                If the `output_type` is "AnnotationStore", the function returns
+                the patch predictor output as an SQLiteStore containing Annotations
+                for each or the Path to a `.db` file depending on whether a
+                save_dir Path is provided. Otherwise, the function defaults to
+                returning patch predictor output, either as a dict or the Path to a
+                `.zarr` file depending on whether a save_dir Path is provided.
+
+        """
+        if (
+            self.cache_mode or not save_dir
+        ) and output_type.lower() != "annotationstore":
+            return processed_predictions
+
+        if output_type.lower() == "zarr":
+            return super().save_predictions(
+                processed_predictions, output_type, save_dir, **kwargs
+            )
+
+        save_path = Path(kwargs.get("output_file", save_dir))
+
+        # scale_factor set from kwargs
+        scale_factor = kwargs.get("scale_factor", (1.0, 1.0))
+        # class_dict set from kwargs
+        class_dict = kwargs.get("class_dict")
+
+        processed_predictions_path: str | Path | None = None
+
+        # Need to add support for zarr conversion.
+        if self.cache_mode:
+            processed_predictions_path = processed_predictions
+            processed_predictions = zarr.open(processed_predictions, mode="r")
+
+        save_paths = []
+
+        for i, predictions in enumerate(processed_predictions["predictions"]):
+            if isinstance(self.images[i], Path):
+                output_path = save_path / (self.images[i].stem + ".db")
+            else:
+                output_path = save_path / (str(i) + ".db")
+
+            out_file = dict_to_store_semantic_segmentor(
+                patch_output={"predictions": predictions},
+                scale_factor=scale_factor,
+                class_dict=class_dict,
+                save_path=output_path,
+            )
+
+            save_paths.append(out_file)
+
+        if processed_predictions_path is not None:
+            shutil.rmtree(processed_predictions_path)
+
+        return save_paths
+
+    def run(
+        self: SemanticSegmentor,
+        images: list[os | Path | WSIReader] | np.ndarray,
+        masks: list[os | Path] | np.ndarray | None = None,
+        labels: list | None = None,
+        ioconfig: IOSegmentorConfig | None = None,
+        *,
+        patch_mode: bool = True,
+        save_dir: os | Path | None = None,  # None will not save output
+        overwrite: bool = False,
+        output_type: str = "dict",
+        **kwargs: Unpack[SemanticSegmentorRunParams],
+    ) -> AnnotationStore | Path | str | dict | list[Path]:
+        """Run the engine on input images.
+
+        Args:
+            images (list, ndarray):
+                List of inputs to process. when using `patch` mode, the
```

Comment:
Capitalise `when`

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/866/commits/322d32e33ae09e7d3b4697070fa12e9d7ad281a1